### PR TITLE
Added HRAM emulation

### PR DESCRIPTION
**Pull request by elialm**
I've gone ahead and added emulation of HRAM. It indeed was little to add. I've also changed the unimplemented memory access panic message to output the address in hex, since that is easier to chew.

I bumped the version to 1.4.0, not sure how you generally handle this so do let me know if you'd like to keep the version number or change it to something else.

### CODE REVIEW

File: test/test.asm [hunk 1]
```diff
@@ -46,3 +52,7 @@ wVariable:
 	db
 
 wString: ds strlen("Hello, world!") + 1
+
+SECTION "HighMemory", HRAM
+hVariable:
+	db
```
Comment (DaKnig) [inline test/test.asm]:
```suggestion
	db

```
please add a newline at the EOF

File: test/test.sh
Comment (DaKnig) [inline test/test.sh:1]: if you wanna add a test and such, maybe make it a separate PR?

**Review by DaKnig — CHANGES_REQUESTED** [review]

**Comment by DaKnig** [comment]
overall, simple, short PR.

File: test/test.sh
Comment (elialm) [reply test/test.sh:1]: Removed in d23f4d1. I'll maybe add it in a separate PR then. It was just a minor automatisation, since I didn't want to write the entire evuint command separately.

File: src/main.rs [hunk 2]
```diff
@@ -347,6 +348,13 @@ fn main() {
 
 	let mut logger = Logger::new(silence_level, &rom_path);
 
+	// create dump dir if it does not exist already
+	if let Some(ref dump_dir) = cli.dump_dir {
+		if let Err(msg) = fs::create_dir_all(dump_dir) {
```
Comment (DaKnig) [inline src/main.rs:352]: maybe add logic to avoid creating a directory if one already exists, to avoid an error. not mandatory.

Comment (elialm) [reply src/main.rs:352]: The `fs::create_dir_all` methods already does that. From its documentation:

    Recursively create a directory and all of its parent components if they are missing.

Comment (elialm) [reply src/main.rs:352]: The error it would have returned is only for other stuff, like insufficient permissions. But handling such issues should be left to the user I think.

Comment (elialm) [reply src/main.rs:352]: One thing I could maybe do is treat failure to create a dump directory as a fatal error and exit the program. I think it would be quite reasonable to do that, since the user explicitly states that he/she wants a dump directory, but the program was not able to create one. That would be a problem imo. What are your thoughts?

Comment (DaKnig) [reply src/main.rs:352]: imo you are right and can close the issue

Comment (elialm) [reply src/main.rs:352]: I've added a call to `exit` then to quit the program on any failure to create the dump dir as discussed. Closing the issue.

**Comment by evie-calico** [comment]
Just made a few small changes; I'm only bumping to 1.3.2 because both of these things are really only bug fixes. Thank you!
